### PR TITLE
[DEVELOP] VLANGroup Ranges

### PR DESCRIPTION
**Pull request by HanlinMiao**
Cherrypicked VLANGroup Range PRs

- [x] https://github.com/nautobot/nautobot/pull/6176/
- [x] https://github.com/nautobot/nautobot/pull/6256
- [x] https://github.com/nautobot/nautobot/pull/6214 (included in #6176)

### CODE REVIEW

File: nautobot/docs/user-guide/core-data-model/ipam/vlangroup.md [hunk 1]
```diff
@@ -3,3 +3,191 @@
 VLAN groups can be used to organize VLANs within Nautobot. Each group may optionally be assigned to a specific location, but a group cannot belong to multiple locations.
 
 Groups can also be used to enforce uniqueness: Each VLAN within a group must have a unique ID and name. VLANs which are not assigned to a group may have overlapping names and IDs (including VLANs which belong to a common location). For example, you can create two VLANs with ID 123, but they cannot both be assigned to the same group.
+
+## VLAN Group Ranges
+
++++ 2.3.0
```
Comment (glennmatthews) [inline nautobot/docs/user-guide/core-data-model/ipam/vlangroup.md:9]: `2.3.6` at this point,  I guess? Same comment below.

File: nautobot/ipam/api/views.py [hunk 2]
```diff
@@ -387,10 +388,171 @@ class IPAddressToInterfaceViewSet(NautobotModelViewSet):
...
+                available_vids = vlan_group.available_vids
+                if len(available_vids) < len(requested_vlans):
+                    return Response(
+                        {
+                            "detail": (
+                                f"An insufficient number of VLANs are available within the VLANGroup {vlan_group} "
+                                f"({len(requested_vlans)} requested, {len(available_vids)} available)"
+                            )
+                        },
+                        status=status.HTTP_204_NO_CONTENT,
```
Comment (glennmatthews) [inline nautobot/ipam/api/views.py:464]: Maybe a followup issue? I realize this is following the same pattern as our `available-prefixes` and `available-ips` endpoints, but a 204 response is definitely not the most intuitive response code, should really probably be a 400 I think. Sorry to have missed this before.

File: nautobot/ipam/models.py [hunk 3]
```diff
@@ -1274,11 +1275,14 @@ def __str__(self):
 
 
 @extras_features(
+    "custom_links",
     "custom_validators",
+    "export_templates",
     "graphql",
     "locations",
+    "webhooks",
```
Comment (glennmatthews) [inline nautobot/ipam/models.py:1283]: Would be good to have a release-note mention of these additions as well.

**Review by glennmatthews — APPROVED** [review]

File: nautobot/ipam/api/views.py (hunk 2 above)
Comment (HanlinMiao) [reply nautobot/ipam/api/views.py:464]: https://github.com/nautobot/nautobot/issues/6310

File: nautobot/docs/release-notes/version-2.3.md [hunk 4]
```diff
@@ -71,6 +71,10 @@ Added [a set of functionality](../user-guide/platform-functionality/objectmetada
 
 Nautobot now supports Python 3.12, and Python 3.12 is now the default Python version included in the `nautobot` Docker images.
 
+#### VLANGroup Model Enhancement ([#6309](https://github.com/nautobot/nautobot/issues/6309))
+
+Added a required `range` field on the `VLANGroup` model with a default value of `1-4094`. `VLANGroup` model now also supports `custom_links`, `export_templates`, `tags`, and `webhooks`.
```
Comment (glennmatthews) [inline nautobot/docs/release-notes/version-2.3.md:76]:
```suggestion
#### VLANGroup Model Enhancement ([#6309](https://github.com/nautobot/nautobot/issues/6309))

+++ 2.3.6

Added a `range` field on the `VLANGroup` model with a default value of `1-4094`. `VLANGroup` model now also supports `custom_links`, `export_templates`, `tags`, and `webhooks`.
```

**Pull request merged**